x/stake/tick_test.go

```diff
@@ -120,14 +127,522 @@ func TestProcessProvisions(t *testing.T) {
+		// every other hour it will just add normal provisions
+		if twoWeekCounter == hr && candidateCounter < 20 {
+
+			//Get values before randomOperation
+			expInflationBefore := keeper.nextInflation(ctx)
+			initialBondedPool := pool.BondedPool
+			initialUnbondedPool := pool.UnbondedPool
+
+			//Random operation, and recording how candidates are modified
+			poolMod, candidateMod, tokens, msg := randomOperation(r)(r, pool, candidates[candidateCounter])
+			candidatesMod := make([]Candidate, len(candidates))
+			copy(candidatesMod[:], candidates[:])
+			require.Equal(t, numCandidates, len(candidates), "i %v", candidateCounter)
+			require.Equal(t, numCandidates, len(candidatesMod), "i %v", candidateCounter)
+			candidatesMod[candidateCounter] = candidateMod
+
+			assertInvariants(t, msg,
+				pool, candidates,
+				poolMod, candidatesMod, tokens)
+
+			//set pool and candidates after the random operation
+			pool = poolMod
+			keeper.setPool(ctx, pool)
+			candidates = candidatesMod
+
+			//Get values after randomOperation
+			expInflationAfter := keeper.nextInflation(ctx)
+			afterBondedPool := pool.BondedPool
+			afterUnbondedPool := pool.UnbondedPool
+
+			//Process provisions after random operation
+			pool = keeper.processProvisions(ctx)
+			keeper.setPool(ctx, pool)
+
+			//Comparing expected inflation before and after the randomOperation()
+			inflationIncreased := expInflationAfter.GT(expInflationBefore)
+			inflationEqual := expInflationAfter.Equal(expInflationBefore)
+
+			if afterBondedPool > initialBondedPool {
+				//Inflation will NOT increase, because we are adding bonded tokens to the pool
+				//CASE: Happens when we randomly add tokens to a bonded candidate
+				//CASE: Happens when we randomly bond a candidate from unbonded
+
+				//for the off case where we are bonded so low (i.e. 15%) , that we add bonded tokens and inflation is unchanged at 20%
+				//for the off case where we are bonded so high (i.e. 90%), that we add bonded tokens and inflation is unchanged at 7%
+				if expInflationBefore.Equal(sdk.NewRat(20, 100)) {
+					assert.True(t, !inflationIncreased || inflationEqual, msg)
+				} else if expInflationBefore.Equal(sdk.NewRat(7, 100)) {
+					assert.True(t, !inflationIncreased || inflationEqual, msg)
+				} else {
+					assert.True(t, !inflationIncreased, msg)
+				}
+
+			} else if afterBondedPool < initialBondedPool {
+				//Inflation WILL increase, because we are removing bonded tokens from the pool
+				//CASE: Happens when we randomly remove bonded Shares
+				//CASE: Happens when we randomly unbond a candidate that was bonded
+
+				//For the off case where we are bonded so low (i.e. 15%),  and more tokens are unbonded, inflation is unchanged at 20%
+				//For the off case where we are bonded so high (i.e. 90%) and we unbond a small amount (maybe 2%), inflation  is unchanged at 7%
+				if expInflationBefore.Equal(sdk.NewRat(20, 100)) {
```

Comment:
We're way to many layers deep, let's break this out into either separate functions, not sure we should have hard coded arbitrary checks for things like two-week counter or 1600